File: dataModuleTester/src/main/java/com/mobnetic/coinguardiandatamodule/tester/volley/generic/ResponseListener.java
package com.mobnetic.coinguardiandatamodule.tester.volley.generic;

import java.util.Map;

import com.android.volley.NetworkResponse;
import com.android.volley.Response.Listener;

public abstract class ResponseListener<T> implements Listener<T> {
	
    public abstract void onResponse(String url, Map<String, String> requestHeaders, NetworkResponse networkResponse, String responseString, T response);
    
    @Override
    public void onResponse(T response) {
    	onResponse(null, null, null, null, response);
    }
}

File: dataModuleTester/src/main/java/com/aneonex/bitcoinchecker/tester/volley/generic/ResponseListener.kt
package com.aneonex.bitcoinchecker.tester.volley.generic

import com.android.volley.NetworkResponse
import com.android.volley.Response

abstract class ResponseListener<T> : Response.Listener<T> {
    abstract fun onResponse(url: String?, requestHeaders: Map<String, String>?, networkResponse: NetworkResponse?, responseString: String?, response: T)
    override fun onResponse(response: T) {
        onResponse(null, null, null, null, response)
    }
}